include/s3select_functions.h

```diff
@@ -2586,6 +2586,17 @@ bool base_statement::is_column_reference() const
   return false;
 }
 
+std::string base_statement::get_key_from_projection()
+{
+  variable* v_name = dynamic_cast<variable*>(this);
+
+  if(v_name)  {
+    return v_name->get_name();
+  } else {
+    return "No key in projection!";
```

Comment:
it returns a string, which will appear in the result.

it should throw an exception.